git/development-workflow.md

```diff
@@ -0,0 +1,224 @@
+the [Merge Strategy](merge-strategy.md) best practice.
+
+#### Push
+
+Next you need to push your local updates to the `origin` remote, so that others
+will see your changes.
+
+If you get an error that a fast-forward merge was not possible, it may mean that
+somebody else pushed commits to the release branch, and you need to fetch the
+latest commits and rebase again.
+
+You should be careful to push your local feature branch (typically using the
+`--force-with-lease` option to overwrite history on the remote) before you push
+your local release branch so that your pull request (or merge request) is
+correctly marked as "merged".
+
+#### Clean up
+
+Finally, your feature is done and merged into the release branch. Delete your
+feature branch, both locally and on the `origin` remote, so that it doesn't
+linger and clutter the list of open branches.
+
+### Appendix: git command reference
+
+So far, we've elided the commands one would use during development, in order not
+to distract from the concepts. We include them here for completeness.
+
+To create a new branch named `feature` that branches from `release`, and
+simultaneously switch to the `feature` branch, say:
+
+```
+git checkout -b feature release
+```
+
+To create commits during development, say `git status` and follow the
+instructions to add and remove files from the index, or staging area for changes
+before they are committed. When you're ready to create a commit from what's in
+the index, say `git commit`. (You may want to set your `$EDITOR` environment
+variable before you do.)
+
+To rewrite history, we recommend reading
+[the relevant section from the Pro Git Book](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History).
+Don't worry too much about messing things up. If you do, you can abort any
+in-progress rebase or, worst case, simply reset your branch to be what's on
+`origin`. Remember that your local changes (including rebases) aren't published
+until you `push`. If you're intimidated, ask for help.
+
+To open a pull request, you can use the web-based UI of GitHub or Gitlab. Or, if
+GitHub is your `origin`, you can use the `hub` command for to open a preview
+page for the pull request. The following instructions assume `git` is aliased to
+`hub`, which is common practice when using `hub` since it passes unrecognized
+commands through to `git`.
+
+```
+git checkout feature        # switch to your feature branch
+git push -u origin feature  # push it to `origin`
+git compare feature         # use the `hub` command to open a comparison webpage
+```
+
+To respond to code review comments, you're usually stuck with the web UI and in
+some cases email replies. If you want to add new commits to your pull request:
+
+```
+# on your `feature` branch
+# do normal development, including `git add` and `git commit`
+git push
+```
+
+To rebase your `feature` branch onto the latest `release` branch:
+
+```
+git checkout release
+git pull
```

Comment:
Hmm, I don't think we've covered that point anywhere. I think it would make more sense in the release branch doc though. I'll update that and open another PR.

If that's alright, is this PR good to go?